src/hotspot/share/logging/logStream.hpp

```diff
@@ -53,60 +59,203 @@ class LogStream : public outputStream {
     void append(const char* s, size_t len);
     void reset();
   };
-  LineBuffer  _current_line;
-  LogTargetHandle _log_handle;
 
-  // Prevent operator new for LogStream.
-  static void* operator new (size_t);
-  static void* operator new[] (size_t);
+private:
+  LineBuffer _current_line;
+protected:
+  BackingLog _backing_log;
+public:
+  explicit LogStreamImpl(BackingLog bl)
+    : _backing_log(bl) {};
+
+  virtual ~LogStreamImpl() {
+    if (_current_line.is_empty() == false) {
```

Comment:
Nit: boolean functions can be used directly in expressions - so `if (!_current_line.is_empty()) {`